app/code/Magento/Catalog/Model/Indexer/Category/Flat/AbstractAction.php

```diff
@@ -81,16 +80,23 @@ class AbstractAction
      * @param ResourceConnection $resource
      * @param StoreManagerInterface $storeManager
      * @param Helper $resourceHelper
+     * @param MetadataPool|null $categoryMetadata
+     * @param SkipStaticColumnsProvider|null $provider
      */
     public function __construct(
         ResourceConnection $resource,
         StoreManagerInterface $storeManager,
-        Helper $resourceHelper
+        Helper $resourceHelper,
+        MetadataPool $categoryMetadata = null,
+        SkipStaticColumnsProvider $provider = null
     ) {
         $this->resource = $resource;
         $this->connection = $resource->getConnection();
         $this->storeManager = $storeManager;
         $this->resourceHelper = $resourceHelper;
+        $this->categoryMetadata = $categoryMetadata ?: ObjectManager::getInstance()->get(MetadataPool::class);
+        $skipProvider = $provider ?: ObjectManager::getInstance()->get(SkipStaticColumnsProvider::class);
+        $this->skipStaticColumns = $skipProvider->get();
```

Comment:
Please do not run getting some data in the constructor. This causes unexpected issues during object creation.

Reply:
@ihor-sviziev I removed it from constructor and returned a method, please check. Thanks!